src/main/java/ua/fan/hw24/entity/Device.java

```diff
@@ -0,0 +1,40 @@
+package ua.fan.hw24.entity;
+
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+import lombok.ToString;
+
+import javax.persistence.*;
+import java.time.LocalDateTime;
+
+@Entity
+@Getter
+@Setter
+@NoArgsConstructor
+@ToString
+public class Device {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Integer id;
+    private  String type;
+    private String modelName;
+    private int price;
+    private LocalDateTime manufactureDate;
+    private String description;
+    private boolean inStock;
+
+    @ManyToOne
+    @JoinColumn(name = "factory_id")
+    private Factory factory;
+
```

Comment:
А удаление ты так и не исправила :) вот такой код лечит удаление, все дело в том, что устройство сохраняется в фабрике
```suggestion
@PreRemove
    private void removeEducationFromUsersProfile() {
        factory.getDevices().removeIf(device -> device.getId().equals(this.id));
    }
```

Reply:
ааа, поняла, спасибо)